commit 826e7b6991428dbf7f35c526392cd5e33c485742
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2019-07-11T16:52:18+00:00 (3 months ago)
    
    OS-7666 vminfod times out while starting with thousands of snapshots

diff --git a/src/vm/node_modules/vmload/vmload-datasets.js b/src/vm/node_modules/vmload/vmload-datasets.js
index 05505eeb..cfe6b320 100644
--- a/src/vm/node_modules/vmload/vmload-datasets.js
+++ b/src/vm/node_modules/vmload/vmload-datasets.js
@@ -298,13 +298,14 @@ function getDatasets(obj, options, callback)
     if (obj.zonepath) {
         assert.strictEqual(obj.zonepath[0], '/');
         parents.push(obj.zonepath.slice(1));
-
+        _options.depth = 1;
     } else {
         /*
          * If we don't have one we're listing all VMs and need to fetch
          * everything anyway.
          */
         listEverything = true;
+        _options.depth = 2;
     }
 
     if (obj.datasets) {
